### PR TITLE
fix(ldap): add connection_timeout to vault_ldap_auth_backend

### LINKED ISSUE #2357: [Enhancement]: vault_ldap_auth_backend connection_timeout
### Description

The resource `vault_ldap_auth_backend` does not support `connection_timeout`, despite it being in the API.  When configured to talk to two LDAP controllers, if the primary fails or is unavailable, Vault does not fail over to the second without this timeout value in place.

### Affected Resource(s) and/or Data Source(s)

* resource_vault_ldap_auth_backend

### Potential Terraform Configuration

```terraform
resource "vault_ldap_auth_backend" "default" {
  connection_timeout = 30
}
```

### References

_No response_

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by nomeelnoj**
Fixes #2357

Signed-off-by: nomeelnoj <nomeelnoj@users.noreply.github.com>

### Description

This PR updates the `vault_ldap_auth_backend` resource to support `connection_timeout`, allowing proper failover to secondary LDAP or domain controllers in the event of a primary outage.

Closes #2357

### Checklist
- [X] Added [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md) entry (only for user-facing changes)
- [X] Acceptance tests where run against all supported Vault Versions

### Output from acceptance testing:

```
╰─ make testacc TESTARGS='-run=testLDAPAuthBackendConfig_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -run=testLDAPAuthBackendConfig_basic -timeout 30m ./...
?   	github.com/hashicorp/terraform-provider-vault	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/cmd/coverage	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/cmd/generate	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/helper	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/internal/consts	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/internal/identity/group	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/internal/identity/mfa	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/internal/pki	[no test files]
ok  	github.com/hashicorp/terraform-provider-vault/codegen	(cached) [no tests to run]
ok  	github.com/hashicorp/terraform-provider-vault/internal/identity/entity	(cached) [no tests to run]
?   	github.com/hashicorp/terraform-provider-vault/internal/sync	[no test files]
?   	github.com/hashicorp/terraform-provider-vault/schema	[no test files]
ok  	github.com/hashicorp/terraform-provider-vault/internal/provider	(cached) [no tests to run]
ok  	github.com/hashicorp/terraform-provider-vault/testutil	(cached) [no tests to run]
ok  	github.com/hashicorp/terraform-provider-vault/util	(cached) [no tests to run]
ok  	github.com/hashicorp/terraform-provider-vault/util/mountutil	(cached) [no tests to run]
ok  	github.com/hashicorp/terraform-provider-vault/vault	0.933s [no tests to run]
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ FEATURES:
 * Update `vault_approle_auth_backend_role_secret_id` to support `num_uses` and `ttl` fields ([#2345](https://github.com/hashicorp/terraform-provider-vault/pull/2345))
 * Add support for `use_annotations_as_alias_metadata` field for the `vault_kubernetes_auth_backend_config` resource ([#2206](https://github.com/hashicorp/terraform-provider-vault/pull/2206))
 * Add support for `allow_empty_principals` field for the `vault_ssh_secret_backend_role` resource ([#2354](https://github.com/hashicorp/terraform-provider-vault/pull/2354))
+* Add support for `connection_timeout` field for the `vault_ldap_auth_backend` resource ([#2358](https://github.com/hashicorp/terraform-provider-vault/pull/2358))
```
Comment (nomeelnoj) [inline CHANGELOG.md:10]: Looks like my auto formatter removed trailing spaces.  Let me know if you want me to revert this file so the git history is cleaner.

File: vault/resource_ldap_auth_backend.go [hunk 2]
```diff
@@ -193,6 +194,11 @@ func ldapAuthBackendResource() *schema.Resource {
 			Computed:  true,
 			Sensitive: true,
 		},
+		consts.FieldConnectionTimeout: {
+			Type:     schema.TypeString,
```
Comment (fairclothjm) [inline vault/resource_ldap_auth_backend.go:198]: Any reason we aren't using a TypeInt like `vault_ldap_secret_backend`:

https://github.com/hashicorp/terraform-provider-vault/blob/d8d1353ce7dbefb81e03fa670a491f5c2b6b8877/vault/resource_ldap_secret_backend.go#L72-L77

File: vault/resource_ldap_auth_backend_test.go [hunk 3]
```diff
@@ -278,7 +279,8 @@ resource "vault_ldap_auth_backend" "test" {
     description            = "example"
     userfilter             = "({{.UserAttr}}={{.Username}})"
     username_as_alias      = true
-    use_token_groups = %s
+    use_token_groups       = %s
+    connection_timeout     = "30s"
```
Comment (fairclothjm) [inline vault/resource_ldap_auth_backend_test.go:283]: We I run this test I get

```
    resource_ldap_auth_backend_test.go:24: Step 1/6 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # vault_ldap_auth_backend.test will be updated in-place
          ~ resource "vault_ldap_auth_backend" "test" {
              ~ connection_timeout      = "30" -> "30s"
                id                      = "tf-test-ldap-path-7969980788853429597"
                # (29 unchanged attributes hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestLDAPAuthBackend_basic (0.54s)
```

File: CHANGELOG.md (hunk 1 above)
Comment (fairclothjm) [reply CHANGELOG.md:10]: Thanks, it may be best to revert the whitespace changes

**Review by fairclothjm — COMMENTED** [review]
@nomeelnoj Thanks for the contribution! Looking good so far. I had a few comments.

File: vault/resource_ldap_auth_backend.go (hunk 2 above)
Comment (nomeelnoj) [reply vault/resource_ldap_auth_backend.go:198]: Funny--i started that way, but when I ran terraform with the custom built binary I got a panic.  I might have missed a configuration, I'll try it again so I can post any specific errors or issues I run into

Comment (nomeelnoj) [reply vault/resource_ldap_auth_backend.go]: I missed a configuration--my fault.  Has been updated to `int` and the test should no longer have the `30s` issue.

**Review by fairclothjm — APPROVED** [review]
Thanks @nomeelnoj !

**Pull request merged**